File: src/main/java/com/product/shoppingdemo/model/Product.java
package com.product.shoppingdemo.model;


import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document("Product")
@Data
public class Product {

    private Product(){
        super();
    }

    @Id
    private Long id;

    private String productName;

    private Integer price;

    private String productDesc;

}

File: shopping-demo/src/app/product.ts
export class Product{
    id: number;
    productName: string;
    price: number;
    productDesc: string;
   
    constructor(id:number = 0, productName: string = '', price:number=0, productDesc: string = ''){
        this.id = id;
        this.productName = productName;
        this.price = price;
        this.productDesc = productDesc;
    }

}